File: test77_out/virtual.mcu/TimerMilli/TimerMilli.js

/*
 * ======== MODULE virtual.mcu.TimerMilli ========
 */

var $$c = function() {
    const TimerMilli = {};
    TimerMilli.TimerMilli = TimerMilli
    TimerMilli.$name = 'virtual.mcu.TimerMilli';
    TimerMilli.pollen$used = 0;
    TimerMilli.targetInit = new $$Ref('virtual_mcu_TimerMilli_targetInit__I');
    TimerMilli.setHandler = new $$Ref('virtual_mcu_TimerMilli_setHandler__E');
    TimerMilli.start = new $$Ref('virtual_mcu_TimerMilli_start__E');
    TimerMilli.stop = new $$Ref('virtual_mcu_TimerMilli_stop__E');
    TimerMilli.isRunning = new $$Ref('virtual_mcu_TimerMilli_isRunning__E');
    TimerMilli.resetTimer = new $$Ref('virtual_mcu_TimerMilli_resetTimer__I');
    TimerMilli.sigAlarm = new $$Ref('virtual_mcu_TimerMilli_sigAlarm__I');
    TimerMilli.setHandlerOnHost = function( h ) {
        var $$text = '';
        (TimerMilli.tickHandler) = (h);
        $$printf("Virtual Hardware Timer Milli setHandlerOnHost()\n");
        return $$text;
    }
    TimerMilli.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    TimerMilli.pollen__uses__ = function() {
    }
    TimerMilli.$$privateInit = function() {
        TimerMilli.repeat = false;
    }
    return TimerMilli;
}
var $$u = $$c();
$units['virtual.mcu.TimerMilli'] = $$u;
$units.push($$u);

File: test77_out/virtual.mcu/TimerMilli/TimerMilli.c

/*
 * ======== string literals ========
 */

static const char virtual_mcu_TimerMilli__s__fileName[] = "TimerMilli.p";
static const char virtual_mcu_TimerMilli__s__0[] = "Virtual Hardware Timer Milli setHandlerOnHost()\n";
static const char virtual_mcu_TimerMilli__s__1[] = "TimerMilli";

/*
 * ======== code injections ========
 */

  
    #include <signal.h>
    #include <sys/time.h>
  

/*
 * ======== function definitions ========
 */

void virtual_mcu_TimerMilli_setHandler__E( virtual_mcu_TimerMilli_HP_handler h ) {
    (virtual_mcu_TimerMilli_tickHandler__V) = (h);
}

void virtual_mcu_TimerMilli_stop__E() {
    (virtual_mcu_TimerMilli_repeat__V) = false;
}

bool virtual_mcu_TimerMilli_isRunning__E() {
    return(  (virtual_mcu_TimerMilli_repeat__V) == true  );
}

void virtual_mcu_TimerMilli_resetTimer__I() {
    if ((virtual_mcu_TimerMilli_repeat__V)) {
         signal(SIGALRM, virtual_mcu_TimerMilli_sigAlarm__I) ;

    }
    
       struct itimerval tout_val;
       tout_val.it_interval.tv_sec = 0;
       tout_val.it_interval.tv_usec = 0;
       tout_val.it_value.tv_sec = 0; 
       tout_val.it_value.tv_usec = 1000;  /* 1 ms second timer */       
       setitimer(ITIMER_REAL, &tout_val,0);
    ;

}

void virtual_mcu_TimerMilli_start__E() {
    virtual_mcu_TimerMilli_resetTimer__I();
     signal(SIGALRM, virtual_mcu_TimerMilli_sigAlarm__I) ;

    (virtual_mcu_TimerMilli_repeat__V) = true;
}

void virtual_mcu_TimerMilli_targetInit__I() {
}

void virtual_mcu_TimerMilli_sigAlarm__I() {
    if ((virtual_mcu_TimerMilli_repeat__V)) {
        virtual_mcu_TimerMilli_resetTimer__I();
    }
    if ((virtual_mcu_TimerMilli_tickHandler__V) != null) {
        virtual_mcu_TimerMilli_tickHandler__V();
    }
}

